plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -238,25 +268,61 @@ impl FirmwareManager {
         smartrest_request: SmartRestFirmwareRequest,
         operation_id: String,
     ) -> Result<(), FirmwareManagementError> {
-        let child_id = smartrest_request.device.as_str();
         let firmware_url = smartrest_request.url.as_str();
         let file_cache_key = digest(firmware_url);
-        let file_transfer_url = format!(
-            "http://{}/tedge/file-transfer/{child_id}/firmware_update/{file_cache_key}",
-            &self.local_http_host
-        );
 
         // <tedge-persistent-root>/cache/<file_cache_key>
         let cache_dest_path = self.get_cache_file_path(&file_cache_key)?;
 
-        self.download_new_file(&cache_dest_path, firmware_url)
+        if cache_dest_path.is_file() {
+            info!(
+                "Hit the file cache={}. File download is skipped.",
+                cache_dest_path.display()
+            );
+            self.handle_firmware_update_request_with_downloaded_file(
+                smartrest_request,
+                operation_id,
+                cache_dest_path,
+            )
             .await?;
+        } else {
+            let download_req = DownloadRequest {
+                url: firmware_url.into(),
+                file_name: file_cache_key,
+            };
+
+            info!("Downloading the firmware from {}", firmware_url);
+            // Send a request to the DownloadManager to download the file asynchornously
+            self.download_req_sndr.send(download_req).await?;
```

Comment:
__To be double-checked__. Since the channel has a buffer for a single message, this will block in case of a concurrent request. This will also block the ` tokio::select!` branch where this `handle_firmware_download_request_child_device` method is called and then block the whole event loop preventing any progress.

Reply:
You were right. I've fixed this by increasing the buffer size but still making sure that only one download request is processed at a time.